lib/irb.rb

```diff
@@ -889,6 +889,10 @@ def IRB.irb_exit(*)
     throw :IRB_EXIT
   end
 
+  def IRB.irb_exit!(*)
+    throw :IRB_EXIT!
```

Comment:
We can pass return value when throwing. So if we do `throw :IRB_EXIT, true` here, I think `Irb#run` can be simplified to:

```rb
      # ....
      force_quit = false

      begin
        force_quit = catch(:IRB_EXIT) do
          eval_input
        end
      ensure
        trap("SIGINT", prev_trap)
        conf[:AT_EXIT].each{|hook| hook.call}
        context.io.save_history if save_history
        Kernel.exit!(0) if force_quit
      end
```